### PR TITLE
Add go tests for GCP stackdriver and pubsub

### LINKED ISSUE #3174: Migrate e2e test for Google Cloud Pub/Sub scaler to Go.
### Proposal

Relates to https://github.com/kedacore/keda/issues/2357, https://github.com/kedacore/keda/issues/2737.

### Use-Case

_No response_

### Anything else?

_No response_

### LINKED ISSUE #3234: Migrate e2e test for GCP StackDriver to Go
### Proposal

Relates to https://github.com/kedacore/keda/issues/2357, https://github.com/kedacore/keda/issues/2737.

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by RamCohen**
Signed-off-by: Ram Cohen <ram.cohen@gmail.com>

Add go tests for GCP stackdriver and pubsub

Fixes #3174 
Fixes #3234

Relates to #2737

### CODE REVIEW

**Comment by JorTurFer** [comment]
@RamCohen  could you please add activation property as part of this PR, we have started to adding this existing scalers as you can see in these PRs:
https://github.com/kedacore/keda/pull/3358
https://github.com/kedacore/keda/pull/3360

**Comment by zroubalik** [comment]
> @RamCohen could you please add activation property as part of this PR, we have started to adding this existing scalers as you can see in these PRs: #3358 #3360

Or this could be contributed as a separate PR if you prefer to do so.

File: tests/scalers_go/gcp_stackdriver/gcp_stackdriver_test.go [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+func testScaleUp(t *testing.T, kc *kubernetes.Clientset) {
+	t.Log("--- testing scale up ---")
+
+	cmd := fmt.Sprintf("%sgsutil cp /usr/lib/google-cloud-sdk/bin/gsutil gs://%s", gsPrefix, bucketName)
+	haveAllReplicas := false
+	for i := 0; i < 60 && !haveAllReplicas; i++ {
+		t.Log("--- upload a file to generate traffic ---")
+		_, err := ExecuteCommand(cmd)
+		assert.NoErrorf(t, err, "cannot upload file to bucket - %s", err)
+		haveAllReplicas = WaitForDeploymentReplicaCount(t, kc, deploymentName, testNamespace, maxReplicaCount, 1, 5)
```
[the remark this answers is not in the document]
Comment (RamCohen) [reply tests/scalers_go/gcp_stackdriver/gcp_stackdriver_test.go]: done

**Comment by RamCohen** [comment]
> > @RamCohen could you please add activation property as part of this PR, we have started to adding this existing scalers as you can see in these PRs: #3358 #3360
> 
> Or this could be contributed as a separate PR if you prefer to do so.

I'd rather do it in a separate PR as it relates to the scaler functionality

**Pull request merged**